doc/man3/OSSL_PARAM_int.pod

```diff
@@ -232,15 +231,16 @@ will be assigned the size the parameter's I<data> buffer should have.
 
 OSSL_PARAM_get_utf8_string() retrieves a UTF8 string from the parameter
 pointed to by I<p>.
-The string is either stored into I<*val> with a length limit of I<max_len> or,
+The string is either stored into I<*val> with a size limit of I<max_len> or,
 in the case when I<*val> is NULL, memory is allocated for the string and
 I<max_len> is ignored.
 If memory is allocated by this function, it must be freed by the caller.
 
 OSSL_PARAM_set_utf8_string() sets a UTF8 string from the parameter pointed to
 by I<p> to the value referenced by I<val>.
 If the parameter's I<data> field is NULL, then only its I<return_size> field
-will be assigned the size the parameter's I<data> buffer should have.
+will be assigned the minimum size the parameter's I<data> buffer should have
+to accomodate the string, including a terminating NUL byte.
```

Comment:
This is not what I see above. There is `p->return_size = len;` in set_string_internal and that does not include the terminating NUL byte in case of UTF8 strings.

Reply:
Where does *len* come from?  A few lines down from that, you have this:

``` C
int OSSL_PARAM_set_utf8_string(OSSL_PARAM *p, const char *val)
{
    if (p == NULL)
        return 0;

    p->return_size = 0;
    if (val == NULL)
        return 0;
    return set_string_internal(p, val, strlen(val) + 1, OSSL_PARAM_UTF8_STRING);
}
```

Reply:
Ah, ok!